python/src/uagents/communication.py

```diff
@@ -0,0 +1,268 @@
+                            )
+                        return MsgStatus(
+                            status=DeliveryStatus.DELIVERED,
+                            detail="Message successfully delivered via HTTP",
+                            destination=envelope.target,
+                            endpoint=endpoint,
+                        )
+                LOGGER.warning(
+                    f"Failed to send message to {envelope.target} @ {endpoint}: "
+                    + (await resp.text()),
+                )
+        except aiohttp.ClientConnectorError as ex:
+            LOGGER.warning(f"Failed to connect to {endpoint}: {ex}")
+        except ValidationError as ex:
+            LOGGER.warning(
+                f"Sync message to {envelope.target} @ {endpoint} got invalid response: {ex}",
+            )
+        except Exception as ex:
+            LOGGER.warning(
+                f"Failed to send message to {envelope.target} @ {endpoint}: {ex}",
+            )
+    LOGGER.error(f"Failed to deliver message to {envelope.target}")
+    return MsgStatus(
+        status=DeliveryStatus.FAILED,
+        detail="Message delivery failed",
+        destination=envelope.target,
+        endpoint="",
+    )
+
+
+async def dispatch_sync_response_envelope(env: Envelope) -> MsgStatus:
+    """Dispatch a synchronous response envelope locally."""
+    await dispatcher.dispatch(
+        env.sender,
+        env.target,
+        env.schema_digest,
+        env.decode_payload(),
+        env.session,
+    )
+    return MsgStatus(
+        status=DeliveryStatus.DELIVERED,
+        detail="Sync message successfully delivered via HTTP",
+        destination=env.target,
+        endpoint="",
+    )
+
+
+async def send_sync_message(
+    destination: str,
+    message: Model,
+    response_type: Type[Model] = None,
+    sender: Identity = None,
+    resolver: Resolver = None,
+    timeout: int = 30,
+) -> Union[Model, JsonStr, MsgStatus]:
+    """
+    Standalone function to send a synchronous message to an agent.
+
+    Args:
+        destination (str): The destination address to send the message to.
+        message (Model): The message to be sent.
+        response_type (Type[Model]): The optional type of the response message.
+        sender (Identity): The optional sender identity (defaults to a generated identity).
+        resolver (Resolver): The optional resolver for address-to-endpoint resolution.
+        timeout (int): The optional timeout for the message response in seconds.
+
+    Returns:
+        Union[Model, JsonStr, MsgStatus]: On success, if the response type is provided, the response
+        message is returned with that type. Otherwise, the JSON message is returned. On failure, a
+        message status is returned.
+    """
+    if sender is None:
+        sender = Identity.generate()
+
+    # TODO: This is not quite right as this should really be resolved by the resolver
+    _, _, destination_address = parse_identifier(destination)
```

Comment:
I only added this to make things work when the envelope is created before using the resolver.